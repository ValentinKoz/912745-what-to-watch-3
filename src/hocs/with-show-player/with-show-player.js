import React, {PureComponent} from "react";

export const withShowPlayer = (Component) => {
  class WithShowPlayer extends PureComponent {
    constructor(props) {
      super(props);

      this.state = {
        showPlayer: false,
      };

      this.handleShowPlayer = this.handleShowPlayer.bind(this);
    }

    handleShowPlayer() {
      this.setState((prevState) => ({showPlayer: !prevState.showPlayer}));
    }

    render() {
      const {showPlayer} = this.state;

      return (
        <Component
          {...this.props}
          showPlayer={showPlayer}
          onShowPlayer={this.handleShowPlayer}
        />
      );
    }
  }
  return WithShowPlayer;
};
