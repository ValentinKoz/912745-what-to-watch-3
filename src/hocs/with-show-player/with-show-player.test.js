import React from "react";
import renderer from "react-test-renderer";
import {withShowPlayer} from "./with-show-player.js";

const MockComponent = () => <div></div>;
const MockComponentWrapped = withShowPlayer(MockComponent);

it(`withShowPlayer is renderer correctly`, () => {
  const tree = renderer
    .create(
        <MockComponentWrapped/>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
