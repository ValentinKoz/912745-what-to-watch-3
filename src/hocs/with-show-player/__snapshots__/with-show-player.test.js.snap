// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withShowPlayer is renderer correctly 1`] = `<div />`;
